### DISCUSSION TITLE
When I draw sub-structure of a polycyclic aromatic, bond angles are not shown correctly.

### DISCUSSION THREAD

**Discussion by mohammad-saber** [rdkit/rdkit#4077] [Q&A]
Hi, thank you for reading my question. 

Assume that I have a polycyclic aromatic (let's call it "parent molecule") as shown below:

```
smile = "c1ccc2ocnc2c1"   
mol = Chem.MolFromSmiles(smile)
```

<img width="271" alt="Screen Shot 2021-04-23 at 9 54 13" src="https://user-images.githubusercontent.com/38423900/115802480-ee0e8a00-a419-11eb-92b6-472e4203855b.png">

When I draw sub-structures of the parent molecule, I notice that the bond angles in sub-structures are different from the bond angles in the parent molecule. Following is the code that I use:

```
from rdkit import Chem
from rdkit.Chem.Draw import rdMolDraw2D
from IPython.display import SVG

smile_1 = 'c(cc)cc'
smile_2 = 'n(co)c(c)c'

m1 = Chem.MolFromSmiles(smile_1,sanitize=False)
Chem.SanitizeMol(m1, sanitizeOps=(Chem.SanitizeFlags.SANITIZE_ALL^Chem.SanitizeFlags.SANITIZE_KEKULIZE^Chem.SanitizeFlags.SANITIZE_SETAROMATICITY))
m2 = Chem.MolFromSmiles(smile_2,sanitize=False)
Chem.SanitizeMol(m2, sanitizeOps=(Chem.SanitizeFlags.SANITIZE_ALL^Chem.SanitizeFlags.SANITIZE_KEKULIZE^Chem.SanitizeFlags.SANITIZE_SETAROMATICITY))

mols = [m1, m2]
smiles = ["smile_1", "smile_2"]

molsPerRow=2
subImgSize=(200, 200)
nRows = len(mols) // molsPerRow
if len(mols) % molsPerRow:
  nRows += 1
  
fullSize = (molsPerRow * subImgSize[0], nRows * subImgSize[1])
d2d = rdMolDraw2D.MolDraw2DSVG(fullSize[0], fullSize[1], subImgSize[0], subImgSize[1])
d2d.drawOptions().prepareMolsBeforeDrawing=False
d2d.DrawMolecules(mols, legends=smiles)
d2d.FinishDrawing()
SVG(d2d.GetDrawingText())
```
  
Which results in the following drawing:

<img width="442" alt="Screen Shot 2021-04-23 at 9 59 12" src="https://user-images.githubusercontent.com/38423900/115802871-ac321380-a41a-11eb-9151-2588c9410b95.png">

As can be seen, the angles between several bonds in sub-structures are different from the parent molecule.  
Is there any way to draw sub-structures with the same bond angles as parent molecule? 
Any help is greatly appreciated.

**Comment by bp-kelley** [comment]
I believe you'll need to compute 2D coordinates for the parent molecule,
then do something like a substructure search and copy the coordinates over.

Let me know if you need an example and I'll see if I can make one.

Cheers,
 Brian

**Reply by mohammad-saber** [reply]
Thank you. I will work on it.
